### PR TITLE
Add Query Tag for Snowflake Operator

### LINKED ISSUE #189: DRAFT: Add querytag to execute method of snowflake Operator
The Openlineage adapter for Snowflake can directly read audit info. To add to the richness of the information available for queries run with Airflow, we want each query to able to be mapped back to an Airflow task. 

To do this, we want the SnowflakeOperator to send back a string that contains `airflow_taskid_dagid_trynumber_TIMESTAMP`  as a [Query Tag](https://docs.snowflake.com/en/sql-reference/parameters.html#query-tag) in the query parameters.

*Acceptance Criteria*
- The Snowflake Operator param `parameters` has a default value of `airflow_taskid_dagid_trynumber_TIMESTAMP`  that is set as a query tag in each query, with a max length of 200 characters.
- This parameter shows up when querying Snowflake's [QUERY_HISTORY](https://docs.snowflake.com/en/sql-reference/functions/query_history.html).

### PR DESCRIPTION
**Pull request by bharanidharan14**
Added Query tag to the snowflake operator by adding this the Openlineage adapter for Snowflake can directly read audit info. To add to the richness of the information available for queries run with Airflow

closes #189

### CODE REVIEW

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 1]
```diff
@@ -56,6 +56,19 @@ def execute(self, context: Context) -> None:
...
         self.log.info("Executing: %s", self.sql)
+
+        default_query_tag = (
+            f"airflow_openlineagens_{self.task_id}_{self.dag.dag_id}_{context['ti'].try_number}"
+        )
+        query_tag = (
+            self.parameters.get("query_tag", default_query_tag) if self.parameters else default_query_tag
+        )
+        session_query_tag = f"ALTER SESSION SET query_tag = '{query_tag}';"
+        if isinstance(self.sql, str):
```
Comment (pankajastro) [inline astronomer/providers/snowflake/operators/snowflake.py:67]: do we also need a unit test for this block?

**Comment by pankajastro** [comment]
Please test the Example DAG as well as Openlineage behaviour once

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 1 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:67]: Yes

**Review by pankajastro — APPROVED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 2]
```diff
@@ -56,6 +56,17 @@ def execute(self, context: Context) -> None:
         By deferring the SnowflakeTrigger class pass along with query ids.
```
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:56]: docstring needs to be updated to reflect additional/ new functionality

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 1 above)
Comment (phanikumv) [reply astronomer/providers/snowflake/operators/snowflake.py:65]: could you point me to the  test for this new functionality

**Pull request closed without merging**